3-0-java-core/3-6-4-random-field-comparator/src/main/java/com/bobocode/se/RandomFieldComparator.java

```diff
@@ -0,0 +1,90 @@
+package com.bobocode.se;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Objects;
+import lombok.SneakyThrows;
+
+/**
+ * A generic comparator that is comparing a random field of the given class. The field is either primitive or
+ * {@link Comparable}. It is chosen during comparator instance creation and is used for all comparisons.
+ * <p>
+ * If no field is available to compare, the constructor throws {@link IllegalArgumentException}
+ *
+ * @param <T> the type of the objects that may be compared by this comparator
+ *
+ * @author Stanislav Zabramnyi
+ */
+public class RandomFieldComparator<T> implements Comparator<T> {
+
+    private final Class<T> targetType;
+    private final Field fieldToCompare;
+
+    public RandomFieldComparator(Class<T> targetType) {
+        Objects.requireNonNull(targetType);
+        this.targetType = targetType;
+        this.fieldToCompare = chooseFieldToCompare(targetType);
+    }
+
+    /**
+     * Compares two objects of the class T by the value of the field that was randomly chosen. It allows null values
+     * for the fields, and it treats null value grater than a non-null value.
+     *
+     * @param o1
+     * @param o2
+     * @return 1 in case of first parameter {@param o1} is greater than second one {@param o2},
+     *         0 if objects are equals,
+     *        -1 in case of first parameter {@param o1} is less than second one {@param o2}.
+     */
+    @Override
+    @SneakyThrows
+    @SuppressWarnings("unchecked")
+    public int compare(T o1, T o2) {
+        Objects.requireNonNull(o1);
+        Objects.requireNonNull(o2);
+        fieldToCompare.setAccessible(true);
+
```

Comment:
When you can to the Comparable, it would be better to force the type. So you know that `value1` and `value2` have the same type comparable type (currently you use Comparable as a raw type). I'm not sure if you know what I mean. I can show it later.